undefined reference to `analogReference'

### Basic Infos

- [x] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [x] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [x] I have tested that the issue is present in current master branch (aka latest git).
- [x] I have searched the issue tracker for a similar issue.
- [x] If there is a stack dump, I have decoded it.
- [x] I have filled out all fields below.

#### Platform

- Hardware:					[**ESP-12**|ESP-01|ESP-07|ESP8285 device|other]
- Core Version:			latest
- Development Env:		[Arduino IDE|**Platformio**|Make|other]
- Operating System:		[Windows|Ubuntu|**MacOS**]

### Settings in IDE

- Module:  			[Generic ESP8266 Module|Wemos D1 mini r2|**Nodemcu**|other]
- Flash Mode:			[qio|dio|other]
- Flash Size: 		[4MB/1MB]
- lwip Variant:   [v1.4|v2 Lower Memory|Higher Bandwidth]
- Reset Method:		[ck|nodemcu]
- Flash Frequency:	[40Mhz]
- CPU Frequency:		[80Mhz|160MHz]
- Upload Using:		[OTA|SERIAL]
- Upload Speed:   [115200|other] (serial upload only)


### Problem Description
`undefined reference to 'analogReference'` when compiling libs that use it.


```PHP
[ Deviot 2.3.0.dev12 ] Starting...
Processing esp12e (platform: espressif8266; board: esp12e; framework: arduino)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/esp12e.html
PLATFORM: Espressif 8266 2.2.2 > Espressif ESP8266 ESP-12E
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
PACKAGES: toolchain-xtensa 2.40802.190218 (4.8.2), tool-esptool 1.413.0 (4.13), tool-esptoolpy 1.20600.0 (2.6.0), framework-arduinoespressif8266 2.20502.0 (2.5.2)
Converting basicntc.ino
LDF: Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 141 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <thermistor> 1.0.2
Compiling .pio/build/esp12e/src/basicntc.ino.cpp.o
Linking .pio/build/esp12e/firmware.elf
/Users/shawn/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: .pio/build/esp12e/lib23e/libthermistor_ID354.a(thermistor.cpp.o):(.text._ZN10THERMISTOR4readEv+0x34): undefined reference to `analogReference'
/Users/shawn/.platformio/packages/toolchain-xtensa/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: .pio/build/esp12e/lib23e/libthermistor_ID354.a(thermistor.cpp.o):(.text._ZN10THERMISTOR4readEv+0x6f): undefined reference to `analogReference'
collect2: error: ld returned 1 exit status
*** [.pio/build/esp12e/firmware.elf] Error 1
========================== [ERROR] Took 5.30 seconds ==========================

[Sat Aug 10 08:51:42 2019]
```

def exists in `arduino.h`

Looks like it was discussed but never implemented?
here https://github.com/esp8266/Arduino/pull/443

--- a/cores/esp8266/core_esp8266_wiring_analog.cpp
+++ b/cores/esp8266/core_esp8266_wiring_analog.cpp
@@ -39,4 +39,16 @@ extern int __analogRead(uint8_t pin)
 
 extern int analogRead(uint8_t pin) __attribute__ ((weak, alias("__analogRead")));
 
+
+void __analogReference(uint8_t mode)
+{
+    // Only DEFAULT is supported on the ESP8266
+    if (mode != DEFAULT) {
+        DEBUGV("analogReference called with illegal mode");
+    }
+}
+
+
+extern void analogReference(uint8_t mode) __attribute__ ((weak, alias("__analogReference")));
+
 };

